src/js/utils/dateUtils.js

```diff
@@ -17,4 +27,149 @@ const dateWithoutTimeZone = ({
   return parsedDate.format(outputDateFormat);
 };
 
+export const parseDateAndStripTimezone = (date, providedFormat) => {
+  const dateWithoutOffset = date.replace(/([+-]\d{2}:\d{2}|Z)$/, '');
+  const formatWithoutOffset = providedFormat.replace('XXX', '');
+  return parse(dateWithoutOffset, formatWithoutOffset, new Date());
```

Comment:
adding this `new Date()` in a few places feels weird - why is it needed to include the current time in some methods?

Reply:
It's needed because of the way the library works. It looks weird, but it's what the doc suggests